content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2416 @@
+  counter reset detection has to be performed between A and B, and another one
+  between C and D. Or both B and D have to be returned with a
+  `CounterResetHint` of `UnknownCounterReset`.
+  
+In summary, whenever the TSDB cannot safely establish that a counter reset
+detection between two samples has happened upon ingestion, it either has to
+perform another counter reset detection or it has to return a
+`CounterResetHint` of `UnknownCounterReset` for the second sample.
+
+Note that there is the possiblity of counter resets that are not detected by
+the procedure described above, namely if the counts in the reset histogram have
+increased quickly enough so that the 1st sample after the counter reset has no
+counts that have decreased compared to the last sample prior to the counter
+reset. (This is also a problem for float counters, where it is actually more
+likely to happen.) With the mechanisms explained above, it is possible to store
+a counter reset even in this case, provided that the counter reset was detected
+by other means. However, due to the complications caused by insertion and
+removal of chunks, out-of-order samples, and overlapping blocks (as explained
+above), this information MAY get lost if a second round of counter reset
+detection is required. (TODO: Currently, this information is reliably lost, see
+TODO above.) A better way to safely mark a counter reset is via created
+timestamps (see next section).
+
+### Created timestamp handling
+
+OpenMetrics introduced so-called created timestamps for counters, summaries,
+and classic counter histograms. (The term is probably short for “created-at
+timstamp”. The more appropriate term might have been “creation timestamp” or
+“reset timestamp”, but the term “created timestamp” is firmly established by
+now.)
+
+The created timestamp provides the most recent time the metric was created or
+reset. A [design
+doc](https://github.com/prometheus/proposals/blob/main/proposals/2023-06-13_created-timestamp.md)
+describes how Prometheus handles created timestamps.
+
+Created timestamps are also useful for native histograms. In the same way a
+synthetic zero sample is inserted for float counters, a zero value of a
+histogram sample is inserted for counter histograms. A zero value of a
+histogram has no populated buckets, and the sum of observations, the count of
+observations, and the zero bucket population are all zero. Schema, zero bucket
+width, custom values, and the float vs. integer flavor of the histogram SHOULD
+match the sample that directly follows the synthetic zero sample (to not
+trigger the detection of a spurious counter reset).
+
+The counter reset information of the synthetic zero sample is always set to
+`CounterReset`. (TODO: Currently, Prometheus probably sets it to
+`UnknownCounterReset` for the first sample of a series, which is not wrong, but
+I think setting it to `CounterReset` makes more sense.)
```

Comment:
Yea, I think we want to switch to native CT handling, not synthetic zeros. So switching counter reset to CT would be one way of doing this, going forward 🤔

Reply:
See above.